lib/overcommit/installer.rb

```diff
@@ -118,6 +125,32 @@ def can_replace_file?(file)
         overcommit_hook?(file)
     end
 
+    def preserve_old_hooks
+      return unless File.directory?(hooks_path)
+
+      ensure_directory(old_hooks_path)
+      Overcommit::Utils.supported_hook_types.each do |hook_type|
```

Comment:
I think that until (if ever) we have comprehensive support for all hook types, we should be very gentle with existing unsupported hooks. That said, I'm not entirely opposed to moving everything to `.git/old-hooks` or similar as long as we explicitly notify the user of this action.

Speaking of which, there should probably be some logger output regardless of what we decide here. I'll throw that in after a bit.